apps/docs/content/guides/auth/auth-mfa/webauthn.mdx

```diff
@@ -0,0 +1,195 @@
+   effort to reduce onboarding friction.
+2. From within a settings page.
+   Allows users to set up, disable or modify their MFA settings.
+
+As far as possible, maintain a generic flow that you can reuse in both cases with minor modifications.
+
+You can enroll a WebAuthn factor with a single step:
+
+```
+import { browserSupportsWebAuthn } from '@supabase/supabase-js'
+if (supabase.auth.browserSupportsWebAuthn()) {
+    const { data, error } = await supabase.auth.mfa.enroll({ factorType: 'webauthn'})
+}) else {
+   // Use another MFA method, such as MFA (Phone)
+   // const { data, error } = await supabase.auth.mfa.enroll({ factorType: 'phone', phone: '<your-phone-number>' })
+};
+```
+
+Or in multiple-steps by setting the `useMultiStepEnroll` parameter to true.
+
+```
+const { data: { factorId }, error } = await supabase.auth.mfa.enroll({ factorType: 'webauthn', useMultiStep: true })
+
+const { data: {options}, error } = await supabase.auth.mfa.challenge({ factorId })
+const options = await supabase.auth.generateRegistrationOptions{
+		options,
+		rp_name,
+		user: {
+			name,
+			display_name,
+		}
+		authenticator_options: {
+			attestation: 'direct', // or developer choice of 'enterprise'
+			authenticatorAttachment,
+			requireResidentKey,
+			userVerification,
+		}
+	}
+})
+
+const credential = await navigator.credentials.create(data)
+const { data, error } = await supabase.auth.mfa.verify({ factorId, credential })
+```
+
+`useMultiStep` informs Auth to abstract away the complexities of interacting with the browser API. Multi Step enrollment is recommended only for Advanced Users who require a heightened level of customization in their application
+
+When using multi-step enroll there are a substantial number of fields to deal with. The fields are highlighted below:
+
+| Field Type       | Field Name                | Configurability                                                                                      | Description                                                           |
+| ---------------- | ------------------------- | ---------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------- |
+| Non-Configurable | `excludeCredentials`      | Generated by auth server                                                                             | Prevents enrollment of multiple credentials                           |
+| Non-Configurable | `rp.id`                   | Generated by auth server                                                                             | Relying Party ID                                                      |
+| Non-Configurable | `challenge`               | Generated by auth server                                                                             | Unique challenge for the authentication attempt                       |
```

Comment:
not sure if the non-configurable options are implemented / enforced at the client lib level but if we don't even show these options in the client lib / accept them in the Auth API, then maybe worth leaving it out